pkg/token/manualtoken_test.go

```diff
@@ -0,0 +1,103 @@
+package token
+
+import (
+	"testing"
+
+	"github.com/Azure/go-autorest/autorest/adal"
+)
+
+func TestNewManualToken(t *testing.T) {
+	testCases := []struct {
+		name        string
+		oAuthConfig adal.OAuthConfig
+		clientID    string
+		resourceID  string
+		tenantID    string
+		token       *adal.Token
+		wantErr     bool
```

Comment:
instead of wanting error or not, can we also verify the returned error matches the expected error? in that case, the wantErr should be of `error` type